pyirf/scripts/make_DL3.py

```diff
@@ -0,0 +1,338 @@
+
+    parser.add_argument(
+        "--config_file",
+        type=str,
+        required=True,
+        help="A configuration file like pyirf/resources/performance.yaml .",
+    )
+
+    parser.add_argument(
+        "--obs_time",
+        type=str,
+        required=True,
+        help="An observation time written as (value.unit), e.g. '50.h'",
+    )
+
+    parser.add_argument(
+        "--pipeline",
+        type=str,
+        required=True,
+        help="Name of the pipeline that has produced the DL2 files.",
+    )
+
+    parser.add_argument(
+        "--debug", action="store_true", help="Print debugging information."
+    )
+
+    args = parser.parse_args()
+
+    # INPUT FROM THE CONFIGURATION FILE
+
+    cfg = load_config(args.config_file)
+
+    # Add obs. time to the configuration file
+    str_obs_time = args.obs_time.split(".")
+    cfg["analysis"]["obs_time"] = {
+        "value": float(str_obs_time[0]),
+        "unit": str(str_obs_time[-1]),
+    }
+
+    # Get input directory
+    indir = cfg["general"]["indir"]
+
+    # Get template of the input file(s)
+    template_input_file = cfg["general"]["template_input_file"]
+
+    # Get output directory
+    outdir = os.path.join(
+        cfg["general"]["outdir"],
+        "irf_{}_Time{}{}".format(  # name could change in future
+            args.pipeline,
+            # cfg["analysis"]["thsq_opt"]["type"], # not needed now
+            # cfg["analysis"]["obs_time"]["value"],
+            # cfg["analysis"]["obs_time"]["unit"],
+            args.obs_time.split(".")[0],
+            args.obs_time.split(".")[-1],
+        ),
+    )  # and create it if necessary
+    if not os.path.exists(outdir):
+        os.makedirs(outdir)
+
+    # =========================================================================
+    #                   READ DL2 DATA AND STORE IT ACCORDING TO GADF
+    # =========================================================================
+
+    # Load FITS data
+    particles = ["gamma", "electron", "proton"]
+    evt_dict = dict()  # Contain DL2 file for each type of particle
+    for particle in particles:
+        if args.debug:
+            print(f"Loading {particle} DL2 data...")
+        infile = os.path.join(indir, template_input_file.format(particle))
+        evt_dict[particle] = read_FITS(config=cfg, infile=infile)
+
+    # =========================================================================
+    #               PRELIMINARY OPERATIONS FOR SPECIFIC PIPELINES
+    # =========================================================================
+
+    # Some pipelines could provide some of the DL2 data in different ways
+    # After this part, DL2 data is supposed to be equivalent, regardless
+    # of the original pipeline.
+
+    # Later we should move this out of here, perhaps under a "utils" module.
+
+    if args.pipeline == "EventDisplay":
+
+        # THETA IS SUPPOSED TO BE IN DL2 DATA ACCORDING TO GADF
```

Comment:
No it is not. The only required columns are:

EVENT_ID, TIME, RA, DEC, ENERGY.

Also, OGADF actually only defines the DL3 level, not DL2.

`THETA` is an optional column and is not what you use here. It is the angular distance from the center of the field of view, not from the source position.

https://open-gamma-ray-astro.github.io/gamma-astro-data-formats/events/events.html

Reply:
ah ok it is not required but optional

anyway remember that this comes from protopipe in which we work only with on-axis simulations (for now), so the source position is the center of the FoV

Reply:
also, probably I am mistaken, but the information on the page you link is indeed also DL2 no? plus maybe something else, but all DL2 information is also there

Reply:
So mainly: the comment is wrong. OGADF does not make any statements about simulated DL2 event lists. 
So I would remove any comment that says that the DL2 is not as required as by OGADF and just calculate what is necessary.

Reply:
I modified it

Reply:
Let me make a proposal: we should define our in-memory data structure we need calculate the IRFs.

Then we define reading functions for different formats in the `io` subpackage, e.g. `read_event_display`, which need to return these data structures.

Reply:
> also, probably I am mistaken, but the information on the page you link is indeed also DL2 no? plus maybe something else, but all DL2 information is also there

OGADF only defines the DL3 data format. it makes no statements about anything related to DL2.

Reply:
So again, I would not focus the sprint on making a script. But to create a small set of well-defined functions with a good API that then can be used by anyone to produce IRF components.

Reply:
removed incorrect comments and modified mapper from user to GADF (which was wrong) to an internal data format protoype